### PR TITLE
Bug 1186638 - Added Breakpad and upload crash reports to crash-stats.mozilla.com

### PR DESCRIPTION
**Pull request by sleroux**
* Adds in https://code.google.com/p/google-breakpad/ for in-process crash reporting and uploads to crash-stats.mozilla.com
* Will replace FXCrashDetector in detecting previous crashes since we can't have multiple signal/uncaught exception handlers set.
* Uploads crash reports to crash-stats.mozilla.com
* Refactored user setting build variables into .xcconfig files for cleaner assignment for breakpad settings

### CODE REVIEW

**Comment by fxios-buildasaur** [comment]
Result of integration 1
Integration took 6 minutes and 0 second.
2 build errors: build-errors

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -298,25 +298,14 @@ class BrowserViewController: UIViewController {
         }
 
         if tabManager.count == 0 && !AppConstants.IsRunningTest {
-            // If we previously crashed, ask the user if they want to restore their tabs
-            if FXCrashDetector.sharedDetector().hasCrashed() {
```
Comment (rnewman) [inline Client/Frontend/Browser/BrowserViewController.swift]: Aww, I liked that dialog.

**Comment by rnewman** [comment]
Request: split this into "add breakpad" and "use breakpad" commits; and see if we can do something less painful than checking in 1000 files of breakpad source.

**Comment by sleroux** [comment]
Will clean up commits and open new PR on different branch.

**Pull request closed without merging**